avocado/core/loader.py

```diff
@@ -647,7 +647,8 @@ def _find_avocado_tests(self, path, class_name=None):
         if os.path.isdir(path):
             path = os.path.join(path, "__init__.py")
 
-        mod = ast.parse(open(path).read(), path)
+        with open(path) as source_file:
+            mod = ast.parse(source_file.read(), path)
```

Comment:
Sure, I need to get used to this... (in py2 it is considered safe, in py3 it's considered ugly and warnings are produced). Btw there are more of these in `avocado/*`, would you be so kind and update all of them?

Reply:
I've fixed the exact same issue in `safeloader.py`, and will address the others (among some style changes too) in a upcoming (WiP) PR.